roles/edpm_neutron_sriov/molecule/default/converge.yml

```diff
@@ -16,7 +16,10 @@
 
 - name: Converge
   hosts: all
+  gather_facts: true
+  become: true
   roles:
     - role: "edpm_neutron_sriov"
       vars:
         edpm_neutron_sriov_agent_config_src: "{{lookup('env', 'MOLECULE_SCENARIO_DIRECTORY')}}/test-data"
+        ansible_user: root
```

Comment:
you should not need to set this too root.

the correct fix is to add these pre_tasks
```
 pre_tasks:
    - name: set basic user fact
      set_fact:
        ansible_user: "{{ lookup('env', 'USER') }}"
      when:
        - ansible_user is undefined
    - name: set basic home fact
      set_fact:
        ansible_user_dir: "{{ lookup('env', 'HOME') }}"
      when:
        - ansible_user_dir is undefined
```

https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_nova/molecule/default/converge.yml#L3-L13

Reply:
> you should not need to set this too root.
> 
> the correct fix is to add these pre_tasks
> 
> ```
>  pre_tasks:
>     - name: set basic user fact
>       set_fact:
>         ansible_user: "{{ lookup('env', 'USER') }}"
>       when:
>         - ansible_user is undefined
>     - name: set basic home fact
>       set_fact:
>         ansible_user_dir: "{{ lookup('env', 'HOME') }}"
>       when:
>         - ansible_user_dir is undefined
> ```
> 
> https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_nova/molecule/default/converge.yml#L3-L13

done

Reply:
recheck

Reply:
ok so there are two things happening

the reason root seams to be need for edpm_neutron_sriov is because its running its molecule tests in prow with the podman driver instead of in zuul with the delegated driver.

we should try to figure out how to fix this because the prow jobs are likely not testing things properly as a result.

however we can proceed with this for now to unblock the nfv work.

Reply:
as an aside i did some digging and 

ansible_user is a connect variable only defined when using the ssh connection type .i.e. not when using podma/kubectl

i belive ansible_user_id is available once we gather facts for all connection types and it contains the username ansible will be using on the remote host. in otherwords i has the same info as ansible_user for ssh connection and also provides that info when using podman/kubectl connections too.

so we should refecator our use of ansible_user to ansible_user_id globally in edpm-ansible as a separate pr